lib/miq_cockpit.rb

```diff
@@ -0,0 +1,133 @@
+require 'uri'
+require 'awesome_spawn'
+module MiqCockpit
+  class WS
+    DEFAULT_PORT = 9000
+
+    attr_accessor(:config_dir)
+
+    def self.url(opts, default_host)
+      opts ||= {}
+
+      # running somewhere else
+      if opts[:external_url]
+        url = URI.parse(opts[:external_url])
+      else
+        url = URI::HTTP.build(:host   => default_host,
+                              :scheme => opts[:ssl] ? "https" : "http",
+                              :port   => opts[:port] || DEFAULT_PORT)
+      end
+
+      # Force all paths to be the same
+      url.path = MiqCockpit::ApacheConfig.url_root
+      url
+    end
+
+    def initialize(opts = {})
+      @opts = opts || {}
+      @config_dir = File.join(File.dirname(__FILE__), '../config')
+    end
+
+    def command(address)
+      args = { :port => @opts[:port] || DEFAULT_PORT.to_s }
+      if address
+        args[:address] = address
+      end
+      args[:'no-tls'] = nil unless @opts[:ssl]
+
+      AwesomeSpawn.build_command_line('/usr/libexec/cockpit-ws', args)
+    end
+
+    def save_config
+      Dir.mkdir(@config_dir) unless File.exist?(@config_dir)
+      Dir.mkdir(File.join(@config_dir, "cockpit")) unless File.exist?(File.join(@config_dir, "cockpit"))
```

Comment:
Or since you are using the full directory later as well...

```ruby
cockpit_dir = File.join(@config_dir, "cockpit")
FileUtils.mkdir_p(cockpit_dir)
fname = File.join(cockpit_dir, "cockpit.conf")